sql/core/src/test/scala/org/apache/spark/sql/JsonFunctionsSuite.scala

```diff
@@ -257,6 +257,18 @@ class JsonFunctionsSuite extends QueryTest with SharedSQLContext {
       "A type of keys and values in map() must be string, but got"))
   }
 
+  test("SPARK-21954: JacksonUtils should verify MapType's value type instead of key type") {
```

Comment:
@viirya, how about puting this test around `to_json unsupported type` here and maybe use Scala function API for consistency?

Reply:
Modified as suggestion.